bin/cron/update_project_count

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env ruby
+#
+#
+
+require 'sequel'
+require File.expand_path('../../../pegasus/src/env', __FILE__)
+require 'cdo/only_one'
+require 'cdo/properties'
+require src_dir 'database'
+
+PEGASUS_DB = Sequel.connect(CDO.pegasus_db_writer.sub('mysql:', 'mysql2:'))
+STORAGE_APPS = PEGASUS_DB[:storage_apps]
+
+# This default number of lines of code is from 2019-1-21.
+DEFAULT_NUMBER_OF_PROJECTS = 35_000_000
+
+def main
+  project_count = 0
+  # Get the current amount
+  STORAGE_APPS.each do |row|
+    if row[:standalone] == true
+      project_count += 1
+    end
+  end
+
+  # round number down to closest million
+  project_count = (project_count / 1_000_000).floor * 1_000_000
+
+  if project_count == 0
+    project_count = DEFAULT_NUMBER_OF_PROJECTS
+  end
+
+  # check if number is bigger than previous count
+  existing_project_count = Properties.get(:metrics)['project_count']
+  if existing_project_count.nil?
+    existing_project_count = DEFAULT_NUMBER_OF_PROJECTS
+  end
+  project_count = [project_count, existing_project_count].max
```

Comment:
More generally, I think it would be simpler to just record the actual project count every time, and then do the rounding when we present it.  That way we remove complicated logic that might or might not increment, we have the actual data available for other purposes going forward, and we can make presentation-time choices on things like rounding strategy.

Reply:
Good question.  All four locations go through a common `fetch_metrics` helper function so perhaps we can have the common work done there.

Reply:
@breville @madelynkasula take a look at the way i refactored this. I pulled it out of the update_project_count script and created a method that does the rounding in properties.rb. is this ok?

Reply:
I might be missing something but I'm not sure why you need to fetch the value, then effectively fetch it again in the round function.

I thought you could just add a new helper function to Properties.rb, say `fetch_project_count`, which fetched the metric once and returned it rounded.  The function could return a hash so that you could put the returned value in a field with name something like `project_count_rounded` to differentiate from other future potentially-returned values, say unrounded.

Reply:
All the fetch in that method was used before as a way to try to check that the new number we got was higher than the old number we had. You are right at this point it doesn't work the way we had intended it to. I'll move that check back over to the update_project_count file